### ISSUE TITLE
fo-dicom throws a FormatException when parsing a culture sensitive string to float

### ISSUE THREAD

**Issue by nirajdoshi1985** [fo-dicom/fo-dicom#1320]
https://github.com/fo-dicom/fo-dicom/blob/78e702ed486cff40df145075b0cdc1afd2f3707b/DICOM/DicomDataset.cs#L1383

**Steps to reproduce**
 1. Initialize a DicomDataset collection
 2. Add DicomTags with VR=FL or VR=FD to the collection with floating point numbers as a string. F​or e.g.,

            dicomDataset.Add(DicomTag.TableTopRollAngle, "0.5"); 
            dicomDataset.Add(DicomTag.TableTopPitchAngle, "0.5");

3. Run the code with a culture format on PC that supports decimal point as a decimal separator.
      - **Outcome:** The code should not throw any runtime exception and the tags will be added to the collection.
4. Change the culture format of the PC to a culture that has a uses a different separator than decimal point. For e.g. SE culture uses comma (,) as their decimal point separator. Run the code.
     - **Outcome:** The code will throw a System.FormatException : Input string was not in a correct format. at runtime.

The code internally uses "float.Parse", and "double.Parse" to parse these strings without any cultural accommodations. For all other data types, fo-dicom directly uses value.Cast<string>, while for float and double they are using the corresponding "Parse" methods.

The above example will pass for the US culture, but will fail for the EU cultures.

Need to verify if this is intentional on fo-dicom's side where they would expect culture sensitive values, or was it just a bug.
One of the solutions could be to send in a culture invariant parser.

![image](https://user-images.githubusercontent.com/15216094/152162583-06de0698-d5c7-44b2-8058-3abc715b8036.png)

**Comment by mrbean-bremen** [comment]
This sounds like a bug to me - in this case indeed invariant culture should be used.

**Comment by gofal** [comment]
difficult topic. Thats what the whole culure thing is about. if you are running your applicaiton on a us-machine, and your software receives some data, then it is supposed to be formated in us culutre.
But if you run your software on a eu-machine, then you are supposed to receive data formatted in eu culture. 
So you are requesting to drop the culuture support in fo-dicom. That would be possible, but requires some discussion

**Comment by nirajdoshi1985** [comment]
According to the DICOM Standard, the Floating Single or Floating Double values conform to IEEE 754:1985 32-bit Floating Point Number Format. Not a 100% sure, but IEEE accepts decimal point "." as a number separator.

If this assumption is correct, and if we enable parsing with InvariantCulture, this could lead to another issue. A number string in EU "1,234", could turn out to be a large number when parsed with InvariantCulture.

**Comment by mrbean-bremen** [comment]
Ah, I misunderstood the issue, sorry. Well, that is a matter of documentation. It is not really a bug, but as a developer I actually would expect the numbers to be in invariant format, though I understand that this can be seen differently.

If you add a string value to a `DS` tag, it would be invalid if it were not in the invariant (aka English) format, so for consistence I would expect the input for FL/FD to have the same format. As a developer accustomized to programming language syntax I would expect the same, though I can see that there could be cases where the data is fed in from another source which has another format.

In short, I'm for changing this, but we should at least clearly document the expected value if we decide not to change it (and if we change it too, of course).

**Comment by mrbean-bremen** [comment]
> According to the DICOM Standard, the Floating Single or Floating Double values conform to IEEE 754:1985 32-bit Floating Point Number Format. 

This standard is about the binary format, not about the string representation. Allowing to set FL/FD using a string value is just a convenience, and the string value has to be converted to an (IEEE conformant) float or double in this case. The DICOM standard is not involved here - how to allow the user to input the value is a decision that has to be made in application code.

**Comment by gofal** [comment]
Yes, the Culture behavior of fo-dicom is currently:
- internally of course the iso standard is used, that is pretty much the same as the englisch culture. Also when serializing to or deserializing from dicomjson. there the standard also defines this. 
- in the api to the user, the current culture is used. both for parsing as for serializing.

So your suggetion would be, to drop support for culture and make fo-dicom cultureinvariant. But that would not only mean every parsing method, but then we would have to make that everywhere for consistency. so when calling a method `dataset.GetString(DicomTag.WindowCenter)' where some decimal value is serialized, we would also have to return a string like "251.5" instead of for example "251,5" on german systems.

**Comment by gofal** [comment]
i guess, changing that would break some applications out there.
Maybe we could now add internally in every '.pare' and every '.tostring' method a CultureInfo. But we would not hardcode some cultureinfo, but would have a static global property that is initialized to CurrentCulture (to have the same behavior as until now). But you could change this static global property on application startup to InvariantCulture. 
Then we would support both, culture aware and culture invariant. And we would extend fo-dicom and not break existing applications. 
Do you see any downside of having such a configuration?

**Comment by mrbean-bremen** [comment]
> But we would not hardcode some cultureinfo, but would have a static global property that is initialized to CurrentCulture (to have the same behavior as until now). But you could change this static global property on application startup to InvariantCulture.

Yes, I think this is a good idea. I also had doubts about compatibilty issues, but this would be a relatively small and upwards-compatible change.

**Comment by nirajdoshi1985** [comment]
@gofal 

That is what I was getting to where I noted the issue with hard-coding Invariant Culture. I agree with your suggestion to provide an option to set the CultureInfo globally in fo-dicom-api.

**Comment by gofal** [comment]
Great. The change would not be too difficult. Would you like to do the change and make a PR?

**Comment by jaime-olivares** [comment]
According to a fast analysis, this is not a heavy lift.
Only three files may be affected by InvariantCulture, two of them related to JSON serialization.
That leaves only DicomElement as the most critical class:

<img width="442" alt="image" src="https://github.com/user-attachments/assets/71256613-df9a-47aa-a620-37621ae0b8bd">
